src/main/java/com/mathworks/ci/MatlabCommandStepExecution.java

```diff
@@ -72,11 +72,10 @@ private synchronized int execMatlabCommand(FilePath workspace, Launcher launcher
 
         try {
             matlabLauncher = getProcessToRunMatlabCommand(workspace, launcher, listener, envVars,
-                    uniqueCommandFile, uniqueTmpFldrName);
-            launcher.launch().pwd(uniqeTmpFolderPath).envs(envVars);
+                    "cd('"+ uniqeTmpFolderPath.getRemote().replaceAll("'", "''") +"');"+ uniqueCommandFile, uniqueTmpFldrName);
             listener.getLogger()
                     .println("#################### Starting command output ####################");
-            return matlabLauncher.pwd(uniqeTmpFolderPath).join();
+            return matlabLauncher.pwd(workspace).join();
```

Comment:
Do you have to set `pwd`? I'm wondering if you don't set it, if it will just properly use the current working folder? Then the user can `cd` in their pipeline and we'll properly launch MATLAB from the folder they are in, rather then assuming they are in the workspace root.

Reply:
its not mandatory to set pwd however i have few questions if we did not set it.
* is there any other place than workspaace that user would keep the code ?
* how user would do CD to required directory ? (Using sh command ? )
* even if user sets the directory before invoking MATLAB we will stil invoke the MATLAB from root (by default if we did not set pwd) and the problem of settings.m not getting picked will remain. 
* I tried using cd before our run and it looks like below.

![image](https://user-images.githubusercontent.com/47204011/87848684-29ff4680-c900-11ea-89de-efa04d53f2a4.png)

Reply:
Looks like the assumption with Jenkins is that each build step resets the pwd to the workspace root, so I think the way you have it is good.

Reply:
what if we dont use rmpath anyways the addpath remains valid for only perticular session of MATLAB and since the temp folder is unique namespace colision is remote posssibility

Reply:
If we don't rmpath we have a leaky abstraction. If anyone calls "path" they see a random temp folder on the path. Likewise is we cd but don't cd back. The function handle approach enables us to issue a command "pwd" or "path" and get the right answer back